[BugFix] empty_database_backup_success (#17217)

srlch opened the pull request:
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #17217

## Problem Summary(Required) ：
Problem:
new privilege framework miss the check code for empty database backup

Solution:
add the check code

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

srlch commented:
run starrocks_fe_unittest

chaoyli commented on fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java:
```diff
@@ -1511,6 +1511,9 @@ public Void visitShowSnapshotStatement(ShowSnapshotStmt statement, ConnectContex
         public Void visitBackupStatement(BackupStmt statement, ConnectContext context) {
             checkSystemRepository(context);
             List<TableRef> tableRefs = statement.getTableRefs();
+            if (tableRefs.size() == 0) {
```
Add the tableName in the log.

srlch replied:
There is no table in database. I have changed the error msg include the dbname

srlch committed:
[BugFix] empty_database_backup_success (#17217)
Problem:
new privilege framework miss the check code for empty database backup

Solution:
add the check code

Signed-off-by: srlch <linzichao@starrocks.com>

srlch commented:
run label

wanpengfei-git commented:
run starrocks_admit_test

srlch commented:
run starrocks_fe_unittest

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 4 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|2|00.00%|[1269, 1270]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|2|2|100.00%|[]|
